test/cocotb/osdtestlib/eval_utility.py

```diff
@@ -4,59 +4,87 @@
+    Provides additional methods to check if a packet which was read contains the
+    correct values
     """
 
     @cocotb.coroutine
-    def evaluate_response(self, dut, response, destination, source, reg_width,
-                          expected_payload):
-        """Compare a received payload with its expectation value and output
+    def evaluate_packet(self, dut, packet, mask, destination, source, type,
+                        subtype, expected_payload):
+        """Compare a packet with its expectation value and output
            the result in the console
         Args:
             dut:                device under test.
-            response:           read response of the debug module.
+            packet:             debug interconnect packet which was returned by
+                                the module.
+            mask:               list of boolean variables indicating which flits
+                                of the payload are to be ignored
             destination:        id of the module which initially was the
                                 destination.
             source:             id of the module which initially was the source.
-            reg_width:          data width of the register the data was read 
-                                from/written to.
+            type:               2 bits identifying the packet as a
+                                register access or a debug event.
+            subtype:            for reg access:    information about word width, 
+                                                   request or response, 
+                                                   reported errors.
+                                for debug packets: number of payload words.
             expected_payload:   expected received payload.
         Returns:
             -
         """
 
         result = 1
 
-        if response[0] != source:
-            # test failed if the response was intended for a different module
-            raise TestFailure("Expected dest to be 0x%x, got 0x%x" %
-                              (source, response[0]))
-            result = 0
+        # the destination header is only available in packets using the new
+        # format
+        if packet.new:
+            if packet.destination != source:
+                # test failed if the response was intended for a different
+                # module
+                raise RegAccessFailedException(message="Expected dest to be \
+                0x%x, got 0x%x" % (source, packet.destination))
+
+        if packet.source != destination:
+            # test failed if the response was not sent from the target
+            raise RegAccessFailedException(message="Expected src to be \
+            0x%x, got 0x%x" % (destination, packet.source))
+
+        if packet.packet_type != type:
+            # test failed if the response was not sent from the target
+            raise RegAccessFailedException(message="Expected type to be %s, \
```

Comment:
no need for an exception in this comparison function, a log message and returning False should work. Exceptions are for handling unexpected program behavior. In this function we want to compare data, a failing comparison is what the function is all about and not unexpected behavior. Therefore, use return instead of throwing an exception.